src/config_waiter.rs

```diff
@@ -126,6 +146,40 @@ mod tests {
         // Advance past the BOOT_IDLE_TIMEOUT, we will now have finished
         sleep(BOOT_IDLE_TIMEOUT + Duration::from_millis(100)).await;
 
-        assert_eq!(poll!(wait_booted.as_mut()), Poll::Ready(Ok(())));
+        assert_eq!(poll!(wait_booted.as_mut()), Poll::Ready(Ok(true)));
+    }
+
+    #[tokio::test(start_paused = true)]
+    async fn test_wait_for_boot_cancel() {
+        let config_store = Arc::new(ConfigStore::new_rw());
+
+        let shutdown_signal = CancellationToken::new();
+        let wait_booted = wait_for_partition_boot(config_store.clone(), 0, shutdown_signal.clone());
+        tokio::pin!(wait_booted);
+
+        // nothing produced yet. Move time right before to the BOOT_MAX_IDLE.
+        sleep(BOOT_MAX_IDLE - Duration::from_millis(100)).await;
+
+        // We haven't marked the boot as complete
+        assert_eq!(poll!(wait_booted.as_mut()), Poll::Pending);
+
+        // Add a check config
+        config_store
+            .write()
+            .unwrap()
+            .add_config(Arc::new(CheckConfig::default()));
+
+        // Move time forward to the BOOT_MAX_IDLE. This will NOT mark the boot as complete sicne we
+        // just produced a config. We will need to wait another 400ms for it to complete
+        sleep(Duration::from_millis(200)).await;
```

Comment:
Probably don't need any of this stuff for this test, since you're just going to cancel it and we just want to validate that cancelation happens

Reply:
Yeah, I guess I wanted to simulator a partial startup and that nothing else goes wrong if we cancel